### PR TITLE
Bug 1839640 - Add the new object type and additional behavior for it

### PR DESCRIPTION
**Pull request by badboy**

### CODE REVIEW

File: mozilla_schema_generator/glean_ping.py [hunk 1]
```diff
@@ -329,7 +329,12 @@ def generate_schema(self, config, generic_schema=False) -> Dict[str, Schema]:
             else:
                 generated = super().generate_schema(new_config)
                 for schema in generated.values():
-                    schema.schema.update(defaults)
+                    # We want to override each individual key with assembled defaults,
+                    # but keep values _inside_ them if they have been set in the schemas.
+                    for key, value in defaults.items():
+                        if key not in schema.schema:
+                            schema.schema[key] = {}
+                        schema.schema[key].update(value)
```
Comment (akkomar) [inline mozilla_schema_generator/glean_ping.py:337]: Why is this needed?
Should we add a test to cover this change?

**Review by akkomar — COMMENTED** [review]
Do we need to update https://github.com/mozilla/mozilla-schema-generator/blob/main/Dockerfile#L27 after https://github.com/mozilla/jsonschema-transpiler/pull/107 lands?

File: mozilla_schema_generator/glean_ping.py (hunk 1 above)
Comment (badboy) [reply mozilla_schema_generator/glean_ping.py:337]: Because in https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/793 we set _some_ values but want to override some of them.

Yes, I should add a test.

**Comment by badboy** [comment]
> Do we need to update [`main`/Dockerfile#L27](https://github.com/mozilla/mozilla-schema-generator/blob/main/Dockerfile?rgh-link-date=2024-02-05T09%3A17%3A28Z#L27) after [mozilla/jsonschema-transpiler#107](https://github.com/mozilla/jsonschema-transpiler/pull/107) lands?

yup

**Review by akkomar — APPROVED** [review]
r+wc:
* Dockerfile needs updating
* before merging we should test schema generation on production data to double check that no existing schemas got changed

**Comment by badboy** [comment]
I changed the Dockerfile to read:

```
RUN cargo install --git https://github.com/mozilla/jsonschema-transpiler
```

ran `docker build -t app:build .`
Then ran:

```
docker run --rm \
  -e MPS_SSH_KEY_BASE64="$(base64 deploy_key_ed25519)" \
  -e MPS_REPO_URL="git@github.com:badboy/mozilla-pipeline-schemas" \
  -e MPS_BRANCH_SOURCE="main" \
  -e MPS_BRANCH_PUBLISH="badboy/generated-schemas" \
  -e MPS_VALIDATE_BQ=false \
  app:build
```

This uses a locally constructed key that I added as a deploy key to my fork of MPS.
This took a moment and then resulted in:

```
Everything up-to-date
```

with nothing pushed to: https://github.com/badboy/mozilla-pipeline-schemas/commits/badboy/generated-schemas/

I then re-ran the `docker run` with `MPS_BRANCH_SOURCE="object-metric-type"` and that pushed a new commit: https://github.com/badboy/mozilla-pipeline-schemas/commit/915b833134e70829dd9c44545a3f0b337ba29c65
The only changes there are the addition of the JSONschema parts for object metric (careful: the current github interface _really_ slows down on that commit)
